### ISSUE TITLE
Implement S3 class & utilities for modifying the codemeta object

### ISSUE THREAD

**Issue by cboettig** [codemeta/codemetar#2]
codemeta includes fields that don't exist in DESCRIPTION (e.g. author ORCIDs).  Currently the interface just writes out a `codemeta.json` file directly, rather than providing an intermediate R object and utility functions for modifying (or extracting) metadata from it.  

Ideally `create_codemeta()` would return an S3 object that could be modified with appropriate helper functions for adding these additional fields.  A clever implementation could include the ability to infer some of these fields from sources other than the DESCRIPTION file (e.g. one could image querying GitHub for a list of contributors, and helping add these contributors to the codemeta.json (and optionally, to the DESCRIPTION?) with appropriate flags (e.g. some GitHub contributors might be `mustBeCited = FALSE`).

**Comment by maelle** [comment]
What fields would have their place here rather than in `usethis`/`desc`? 🤔 

I like the idea of querying contributors fro GitHub to add them to DESCRIPTION 😍

**Comment by maelle** [comment]
Related to #85 

But I think `create_codemeta` should be a wrapper around all operations?

Now ORCID can live in DESCRIPTION.

Related to infer contributors from GitHub and adding them to DESCRIPTION see my gist https://gist.github.com/maelle/75261a0b6793d18cfdd02734d2e3f243 mentioned in #95 That said I'm not sure that code using `ghrecipes`/any GH API code & `desc` should live in `codemetar`? I'm a bit biased but I was imagining such a helper living in [`rodev`](https://github.com/ropenscilabs/rodev) unless you think it'd be a general need for developers. Some developers refuse PR before people add themselves as contributors after all.

**Comment by maelle** [comment]
The reason why I think `create_codemeta` should be a wrapper around operations is that otherwise the update workflow of codemeta.json will be a pain.

**Comment by cboettig** [comment]
Yeah, I think I see what you mean, but I'm worried about fully automating look ups that might not be fully reliable -- I think this really needs user confirmation that the name is correct.  (After all, if we could fully automate identifying authors from the existing metadata we wouldn't need ORCID ids to disambiguate authors, right?)  

I agree about better to have the GH API stuff in `rodev` rather then here.  

Re S3 class issue, I'm not sure we need it;  I guess we still have a bit of an open question whether it makes sense to have a native R object representation of the whole 'codemeta.json' file and whether it makes sense to be able to add stuff directly to said file "on the fly" so to speak, rather than requiring that all such metadata live somewhere on disk already (i.e. DESCRIPTION, README).

**Comment by maelle** [comment]
Ah sorry I mean the lookup would help creating a file of some sort manually and for once, and that file would be used automatically
